iec104.cpp

```diff
@@ -963,11 +1033,21 @@ void IEC104Client::m_addData(vector<Datapoint*>& datapoints, int64_t ioa,
         if (feature.value() == "ioa")
             measure_features->push_back(m_createDatapoint(feature.key(), ioa));
         else if (feature.value() == "value")
-            measure_features->push_back(
-                m_createDatapoint(feature.key(), value));
+        {
+            if (typeid(valuestate).name() == typeid(bool).name())
```

Comment:
m_addData is a private function, that is called by two public function. In each; value take the type int64_t or float. 
So why do we test if the type of value is bool ?